src/main/java/org/apache/commons/compress/archivers/zip/ZipFile.java

```diff
@@ -638,13 +646,11 @@ public InputStream getInputStream(final ZipArchiveEntry ze)
         }
         // cast validity is checked just above
         ZipUtil.checkRequestedFeatures(ze);
-        final long start = getDataOffset(ze);
 
         // doesn't get closed if the method is not supported - which
         // should never happen because of the checkRequestedFeatures
         // call above
-        final InputStream is =
-            new BufferedInputStream(createBoundedInputStream(start, ze.getCompressedSize())); //NOSONAR
+        final InputStream is = new BufferedInputStream(getRawInputStream(ze)); //NOSONAR
         switch (ZipMethod.getMethodByCode(ze.getMethod())) {
```

Comment:
This replaces duplicate code with just requesting a raw compressed stream, which should be always available.